components/kyma-environment-broker/internal/edp/client.go

```diff
@@ -111,40 +119,65 @@ func (c *Client) DeleteMetadataTenant(name, env, key string) error {
 
 	response, err := c.httpClient.Do(request)
 	if err != nil {
-		return errors.Wrap(err, "while requesting about delete metadata")
+		return kebError.AsTemporaryError(err, "while requesting about delete metadata")
 	}
 
-	return c.processResponse(response, true)
+	err = c.processResponse(response, true)
+	if err != nil {
+		return errors.Wrap(err, "while processing delete MetadataTenant response")
+	}
+
+	if err := c.closeResponseBody(response); err != nil {
+		return kebError.AsTemporaryError(err, "while closing delete MetadataTenant response")
+	}
+
+	return nil
 }
 
-func (c *Client) GetMetadataTenant(name, env string) (_ []MetadataItem, err error) {
-	response, err := c.httpClient.Get(c.metadataTenantURL(name, env))
+func (c *Client) GetMetadataTenant(name, env string) ([]MetadataItem, error) {
+	var metadata []MetadataItem
+	request, err := http.NewRequest(http.MethodGet, c.metadataTenantURL(name, env), nil)
 	if err != nil {
-		return []MetadataItem{}, errors.Wrap(err, "while requesting about dataTenant metadata")
+		return metadata, errors.Wrap(err, "while creating GET metadata tenant request")
+	}
+	response, err := c.httpClient.Do(request)
+	if err != nil {
+		return metadata, kebError.AsTemporaryError(err, "while requesting about dataTenant metadata")
 	}
-	defer func() {
-		err = multierror.Append(err, errors.Wrap(c.closeResponseBody(response), "while trying to close body reader")).ErrorOrNil()
-	}()
```

Comment:
removing a `defer` may cause not closing the response, for examle when there is a response with broken JSON

Reply:
I removed `defer` but in each client before `return` and after handle the response, I close `response.Body` treat error from this operation as a temporary, e.g.:
```go
if err := c.closeResponseBody(response); err != nil {
    return kebError.AsTemporaryError(err, "while closing delete MetadataTenant response")
}
```
it is because when `multierror` is used we lose the temporary error kind, e.g. in this solution temporary case will not work:
```go
func main() {
	err := test()

	if err != nil {
		switch {
		case IsTemporaryError(err):
			fmt.Println("[TEMP] error ", err)
		default:
			fmt.Println(err)
		}
	}
}

func test() (err error) {
	err = test2()
	defer func() {
		err = multierror.Append(err, errors.Wrap(closeResponseBody(), "while trying to close body reader")).ErrorOrNil()
	}()
	if err != nil {
		return errors.Wrap(err, "while test")
	}
	return nil
}

func test2() (err error) {
	err = test3()
	if err != nil {
		return AsTemporaryError(err, "while test3")
	}
	return nil
}

func test3() error {
	return fmt.Errorf("something bad")
}

func closeResponseBody() error {
	return fmt.Errorf("cannot close")
}
```